lua/telescope/builtin/internal.lua

```diff
@@ -634,26 +634,90 @@ internal.buffers = function(opts)
 end
 
 internal.colorscheme = function(opts)
+  local before_color = vim.api.nvim_exec('colorscheme', true)
+  local need_restore = true
+
   local colors = vim.list_extend(opts.colors or {}, vim.fn.getcompletion('', 'color'))
 
-  pickers.new(opts,{
-    prompt = 'Change Colorscheme',
+  local bufnr = vim.api.nvim_get_current_buf()
+  local p = vim.api.nvim_buf_get_name(bufnr)
+
+  local previewer
+  if vim.fn.buflisted(bufnr) ~= 1 then
+    local deleted = false
+    local function del_win(win_id)
+      if win_id == nil or not vim.api.nvim_win_is_valid(win_id) then
+        return
+      end
+
+      local bnr = vim.api.nvim_win_get_buf(win_id)
+      if vim.api.nvim_buf_is_valid(bnr) and not vim.api.nvim_buf_get_option(bnr, 'buflisted') then
+        vim.cmd(string.format('silent! bdelete! %s', bnr))
```

Comment:
We have a util function for that `utils.buf_delete(bufnr)`

https://github.com/nvim-telescope/telescope.nvim/blob/37a3a68a78e15900254ae5d9d4f9f0b5100a1735/lua/telescope/utils.lua#L338-L350